integration/solana/simple_collectible.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// DISCLAIMER: This file is an example of how to mint and transfer NFTs on Solana. It is not production ready and has not been audited for security.
+// Use it at your own risk.
+
+import '../../solana-library/spl_token.sol';
+
+contract SimpleCollectible {
+    // On Solana, an associated token account saves the amount of tokens an owner has.
+    // It also holds the mint account and the owner account public keys.
+    // The associated token account and the mint account save all the information necessary to prove the authenticty of an NFT.
+    address private associatedTokenAccount;
```

Comment:
Rather than associating a token account, you're probably better off associating the mint.  People can transfer the NFT outside of this program, which will break this connection.

On the flip side, that means less functionality in this contract, since you won't have `transferOwnership`, `getNftOwner`, and `isOwner`.  If you want more functionality here, you can have an additional `metadataAuthority` address, which is allowed to update the uri if they sign.  What do you think?

Reply:
That was a great idea. I've added the metadata authority as you suggested.